File: frogment/src/main/java/ch/grze/frogment/core/component/AbstractFragmentComponentInjector.java
package ch.grze.frogment.core.component;

import android.support.v4.app.Fragment;

import ch.grze.frogment.core.Core;

public abstract class AbstractFragmentComponentInjector {
    abstract public void inject(Core core, Fragment fragment);

    protected <T> T getTypedFragment(Fragment fragment) throws ClassCastException {
        return (T) fragment;
    }
}

File: frogment/src/main/java/ch/grze/frogment/core/component/AbstractFragmentComponentInjector.kt
package ch.grze.frogment.core.component

import android.support.v4.app.Fragment

@Deprecated("fallback, use ComponentInjector")
abstract class AbstractFragmentComponentInjector : ComponentInjector<Fragment> {
    @Throws(ClassCastException::class)
    protected fun <T> getTypedFragment(fragment: Fragment): T = fragment as T
}
